pkg/instancegroups/rollingupdate.go

```diff
@@ -72,10 +74,16 @@ type RollingUpdateCluster struct {
 
 	// ValidateCount is the amount of time that a cluster needs to be validated after single node update
 	ValidateCount int
+
+	Ctx context.Context
```

Comment:
actually there is no need to embed the context within this struct. We can propagate it at every step.

Reply:
I got tired of passing on this one all the time :p what is the benefit of keep propagating it?

Reply:
I mean, the doc states not to embed a context.Context within a struct. However, I found https://github.com/golang/go/issues/22602 which points out that if it is a parameter struct, i.e. to not get a parameter explosion within the function then it would be fine. In our case it is not really only a parameter struct so I'd prefer to keep the context as a first parameter.

Reply:
It is not strictly a parameter struct since we define functions on it, but none of the functions mutate the struct, so for all intents and purposes it is. The concerns in that issue is not applicable to this struct either as far as I can tell.